src/js/components/TextInput/stories/tag-input.stories.js

```diff
@@ -0,0 +1,149 @@
+  const tag = (
+    <Box
+      direction="row"
+      align="center"
+      background="brand"
+      pad={{ horizontal: 'xsmall', vertical: 'xxsmall' }}
+      margin={{ vertical: 'xxsmall' }}
+      round="medium"
+      {...rest}
+    >
+      <Text size="xsmall" margin={{ right: 'xxsmall' }}>
+        {children}
+      </Text>
+      {onRemove && <FormClose size="small" color="white" />}
+    </Box>
+  );
+
+  if (onRemove) {
+    return <Button onClick={onRemove}>{tag}</Button>;
+  }
+  return tag;
+};
+
+class TagInput extends Component {
+  state = {
+    currentTag: '',
+  };
+
+  boxRef = createRef();
+
+  componentDidMount() {
+    this.forceUpdate();
+  }
+
+  updateCurrentTag = event => this.setState({ currentTag: event.target.value });
+
+  onAddTag = tag => {
+    const { onAdd } = this.props;
+    if (onAdd) {
+      onAdd(tag);
+    }
+  };
+
+  onEnter = () => {
+    const { currentTag } = this.state;
+    if (currentTag.length) {
+      this.onAddTag(currentTag);
+      this.setState({ currentTag: '' });
+    }
+  };
+
+  renderValue = () => {
+    const { value, onRemove } = this.props;
+    return value.map((v, index) => (
+      <Tag margin="xxsmall" key={`${v}${index}`} onRemove={() => onRemove(v)}>
+        {v}
+      </Tag>
+    ));
+  };
+
+  render() {
+    const { value = [], onAdd, onRemove, onChange, ...rest } = this.props;
+    const { currentTag } = this.state;
+    return (
+      <Keyboard onEnter={this.onEnter}>
+        <Box
+          direction="row"
+          align="center"
+          pad={{ horizontal: 'xsmall' }}
+          border="all"
+          ref={this.boxRef}
+          wrap
+        >
+          {value.length > 0 && this.renderValue()}
+          <Box flex style={{ minWidth: '120px' }}>
+            <TextInput
+              type="search"
+              plain
+              dropTarget={this.boxRef.current}
+              {...rest}
+              onChange={this.updateCurrentTag}
+              value={currentTag}
+              onSelect={event => {
+                event.stopPropagation();
+                this.onAddTag(event.suggestion);
+              }}
+            />
+          </Box>
+        </Box>
+      </Keyboard>
+    );
+  }
+}
+
+class TagTextInput extends Component {
+  state = {
+    selectedTags: ['foo', 'sony'],
+  };
+
+  onRemoveTag = tag => {
+    const { selectedTags } = this.state;
+
+    const removeIndex = selectedTags.indexOf(tag);
+
+    const newTags = [...selectedTags];
+    if (removeIndex >= 0) {
+      newTags.splice(removeIndex, 1);
+    }
+    this.setState({
+      selectedTags: newTags,
+    });
+  };
+
+  onAddTag = tag => {
+    const { selectedTags } = this.state;
+    this.setState({
+      selectedTags: [...selectedTags, tag],
+    });
+  };
+
+  render() {
+    const { selectedTags } = this.state;
+    return (
+      <Grommet full theme={grommet}>
+        <Box pad="small">
+          <TagInput
+            placeholder="Search for aliases..."
```

Comment:
It might be a little confusing to use the term `Search` in this context since it is not a real search box and the user can't filter out options by entering text on the TextInput.

Reply:
what do you mean by that? at least it is supposed to be a real search box and it is supposed to filter the suggestions.

Reply:
I guess the way I associate `Search` is with a filter functionality, and the filter doesn't seem to work since all the options are displayed no matter what you enter on the search box, makes sense?
![Screen Shot 2019-03-22 at 2 06 01 PM](https://user-images.githubusercontent.com/6320236/54850133-e49b1c80-4cab-11e9-8bb2-e04a48777e1e.png)

Reply:
oh yeah that's a bug. we need to fix it

Reply:
I will look into it.